MSProject/ViewControllers/LibraryTableView/ViewModel/WBLibraryViewModel.swift

```diff
@@ -18,50 +20,35 @@ enum SortMethod {
 
 class WBLibraryViewModel {
     
-    private var libraryItems: [WBBook] = []
-
-    private var bookViewModels: [WBBookViewModel] = [] {
-        didSet {
-            reloadViewClosure?()
-        }
+    var libraryItems: [WBBook] = []
+//    private let libraryItems = MutableProperty<[WBBook]>([])
+    private var bookViewModels: [WBBookViewModel] = []
+    
+    let repository: WBBooksRepository
+    
+    init(booksRepository: WBBooksRepository) {
+        repository = booksRepository
     }
     
     var numberOfCells: Int {
         return bookViewModels.count
     }
-    
-    var reloadViewClosure: (() -> Void)?
-    var showAlertClosure: ((Error) -> Void)?
 
     func getCellViewModel(at indexPath: IndexPath) -> WBBookViewModel {
         return bookViewModels[indexPath.row]
     }
     
-    func loadBooks() {
-        let successBooks: ([WBBook]) -> Void = { (books) in
-            self.libraryItems = books
-            self.sortBooks()
-        }
-        
-        let failureBooks: (Error) -> Void = { (error) in
-            self.showAlertClosure?(error)
-        }
-        
-        WBNetworkManager.manager.fetchBooks(onSuccess: successBooks, onError: failureBooks)
-    }
-    
     func selectBook(at indexPath: IndexPath) -> WBBookViewModel {
```

Comment:
I would call this function getBook(at:) selectBook sounds like it produces some side effects

Reply:
sorry, its duplicated code. should use the getCellViewModel(at:) method